tests/chainer_tests/functions_tests/normalization_tests/test_group_normalization.py

```diff
@@ -63,9 +63,21 @@ def setUp(self):
                 {'atol': 1e-2, 'rtol': 1e-2})
 
     def generate_inputs(self):
-        x = numpy.random.uniform(-1, 1, self.shape).astype(self.dtype)
-        gamma = numpy.random.uniform(-1, 1, self.shape[1]).astype(self.dtype)
-        beta = numpy.random.uniform(-1, 1, self.shape[1]).astype(self.dtype)
+        shape = self.shape
+
+        # sample x such that x.std >= min_std
+        min_std = 0.2 if self.dtype == numpy.float16 else 0.02
+        retry = 0
+        while True:
+            x = numpy.random.uniform(-1, 1, shape).astype(self.dtype)
+            x_groups = x.reshape(shape[0], self.groups, -1)
+            if x_groups.std(axis=2).min() >= min_std:
+                break
+            retry += 1
+            assert retry <= 10, 'Too many retries to generate inputs'
```

Comment:
I find this fine, I am just curious about how many retries does it take on average to sample a good input :).

Thanks!

Reply:
For example, `shape=(3, 20), groups=4` samples 5 per group.  For each group std ≥ 0.2 with probability ~97.8%.
```
>>> (np.random.uniform(-1, 1, (1000000, 5)).std(axis=1) > 0.2).sum()
978456
```
A try satisfies the condition for all groups with probability at least `0.97 ** (3 * 4)`.  Thus it fails 10 tries with probability
```
>>> (1 - 0.97 ** (3 * 4)) ** 10
7.235207665894931e-06
```

... I see.  It's not sufficiently small.

Reply:
Thanks a lot for the great explanation!